docs/dorgeclipseopenj9criuimmutableenvvars.md

```diff
@@ -0,0 +1,52 @@
+<!--
+* Copyright (c) 2017, 2023 IBM Corp. and others
+*
+* This program and the accompanying materials are made
+* available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution and is available at
+* https://www.eclipse.org/legal/epl-2.0/ or the Apache
+* License, Version 2.0 which accompanies this distribution and
+* is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* This Source Code may also be made available under the
+* following Secondary Licenses when the conditions for such
+* availability set forth in the Eclipse Public License, v. 2.0
+* are satisfied: GNU General Public License, version 2 with
+* the GNU Classpath Exception [1] and GNU General Public
+* License, version 2 with the OpenJDK Assembly Exception [2].
+*
+* [1] https://www.gnu.org/software/classpath/license.html
+* [2] https://openjdk.org/legal/assembly-exception.html
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
+* Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# -Dorg.eclipse.openj9.criu.ImmutableEnvVars
+
+
+This system property enables adding of environmental variables that the VM cannot change on restore.
+
+:fontawesome-solid-triangle-exclamation:{: .warn aria-hidden="true"} **Restrictions:** This option takes effect only when the `-XX:+EnableCRIUSupport` option is enabled. This option can be used only on restore.
+
+## Syntax
+
+        -Dorg.eclipse.openj9.criu.ImmutableEnvVars=<environment_variables>
+
+Default environment variables are `LANG`, `LC_ALL`, and `LC_CTYPE`. You can specify a list of immutable environment variables separated by a comma. For example,
+
+        -Dorg.eclipse.openj9.criu.ImmutableEnvVars=ENV_VAR1,ENV_VAR2,ENV_VAR3...ENV_VARN
+
+The default set of environment variables cannot be removed from the list and thus can never be changed post-restore.
+
+## Explanation
+
+The VM does not allow the environment variables to change between checkpoint and restore. At the time of restore, the VM can identify only the immutable environment variables that were used before the checkpoint was taken. Any new environment variables set after the checkpoint and before restore are not visible to the VM. If the VM needs new environment variables on restore, you can add the new immutable variables to the `-Dorg.eclipse.openj9.criu.ImmutableEnvVars` list that is saved in a file set by `CRIUSupport.registerRestoreEnvFile()`. The VM reads from the saved files after the process is restored. These environment variables will not be allowed to change through `CRIUSupport.registerRestoreEnvFile()` upon restore.
```

Comment:
I think the general explanations here and earlier in the topic need clarification. I've read the issues, the API, and the blog post and it's still not clear to me how this works. For example, the blog post talks about mutable and immutable sets of env vars (but doesn't mention -Dorg.eclipse.openj9.criu.ImmutableEnvVars). The [API](https://eclipse-openj9.github.io/openj9-docs/api-jdk11/) says "All environment variables specified in the envFile are added as long as they do not modifiy pre-existeing environment variables", which suggests that there aren't mutable env vars. The doc issue says "the JVM only returns environment variables that are immutable pre-checkpoint" but the API doc says "All pre-existing (environment variables from checkpoint run) env vars are retained."

Reply:
@tajila @vijaysun-omr - Could you please provide clarity? Thanks!